Debian Bug #193634:

Some MOD files are distributed with the extensions prepended to their
names, rather than appended.  I think this may be Amiga convention in
general (is it?), but seems to be noticeable on Aminet in particular.
For instance, I downloaded a file called 's3m_ramagard.lha' on Aminet,
with the single file in the archive called 'S3M.Rama Gardens'.

diff -Ndurp mikmod-3.1.6.orig/src/marchive.c mikmod-3.1.6/src/marchive.c
--- mikmod-3.1.6.orig/src/marchive.c	2003-06-02 03:25:28.000000000 +0200
+++ mikmod-3.1.6/src/marchive.c	2003-06-02 03:24:17.000000000 +0200
@@ -218,11 +218,26 @@
 	NULL
 };
 
+/* On the Amiga (Aminet) the module type seems often to be used as prefix */
 static CHAR* prefixmodulepatterns[]={
+	"669.*",
+	"[Aa][Mm][Ff].*",
+	"[Aa][Pp][Uu][Nn].*",
+	"[Dd][Ss][Mm].*",
+	"[Ff][Aa][Rr].*",
+	"[Gg][Dd][Mm].*",
+	"[Ii][Mm][Ff].*",
+	"[Ii][Tt].*",
 	"[Mm][Ee][Dd].*",
 	"[Mm][Oo][Dd].*",
+	"[Mm][Tt][Mm].*",
 	"[Nn][Ss][Tt].*",
-	"[Xx][Mm].*",		/* found on Aminet */
+	"[Ss]3[Mm].*",
+	"[Ss][Tt][Mm].*",
+	"[Ss][Tt][Xx].*",
+	"[Uu][Ll][Tt].*",
+	"[Uu][Nn][Ii].*",
+	"[Xx][Mm].*",
 	NULL
 };
 
